### PR TITLE
simplify dialog (#166)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
after discussion with Florian

### CODE REVIEW

File: gui/doc/dialog.csv [hunk 1]
```diff
@@ -1,25 +1,19 @@
...
-If <i>on_cancel</i> is empty, the button is not shown."
-cancel_label,str,"""cancel""",The text of the cancel button.
-on_validate,str,"""validate""","Name of a function that is triggered when the validate button is pressed.
-The parameters of that function are all optional:
+<li>payload</li>
 <ul>
-<li>gui instance</li>
-<li>id</li>
-<li>action</li>
+<li>args: List with first element containing the index of the label selected.</li>
```
Comment (FabienLelaquais) [inline gui/doc/dialog.csv]: => "List where the first element contains the index of the selected label"

File: gui/doc/dialog.csv [hunk 2]
```diff
@@ -1,25 +1,19 @@
...
+</ul>
 </ul>
-
-If <i>on_validate</i> is empty, the button is not shown."
-validate_label,str,"""Validate""",The text of the validate button.
+"
+close_label,str,"""Close""","The tooltip of the top right close icon button. In the _on_action_ function, args will hold -1."
+labels, str|list[str],None,"A list of labels to show as buttons at the bottom of the dialog. The index of the button in the list is reported as args in the _on_action_ function (-1 for the close icon)."
```
Comment (FabienLelaquais) [inline gui/doc/dialog.csv]: to show in a row of buttons

File: gui/src/components/Taipy/Dialog.tsx [hunk 3]
```diff
@@ -60,13 +57,24 @@ const Dialog = (props: DialogProps) => {
...
 
-    const handleValidate = useCallback(() => {
-        dispatch(createSendActionNameAction(id, tp_onValidate));
-    }, [dispatch, id, tp_onValidate]);
+    const labels = useMemo(() => {
+        if (props.labels) {
+            try {
+                return JSON.parse(props.labels) as string[];
+            } catch (e) {
+                console.info(`Error while parsing dialog.labels\n${(e as Error).message || e}`);
```
Comment (FabienLelaquais) [inline gui/src/components/Taipy/Dialog.tsx]: 'while' is useless, I guess...

File: gui/src/components/Taipy/Dialog.tsx [hunk 4]
```diff
@@ -109,16 +112,15 @@ const Dialog = (props: DialogProps) => {
...
-                    <Button onClick={handleClose} disabled={!active}>
-                        {cancelLabel}
-                    </Button>
-                )}
-                <Button onClick={handleValidate} autoFocus disabled={!active}>
-                    {validateLabel}
-                </Button>
-            </DialogActions>
+            {labels.length ? (
+                <DialogActions>
```
Comment (FabienLelaquais) [inline gui/src/components/Taipy/Dialog.tsx:116]: Too easy, React, hu?

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
I am very surprised not to see 'dialog.md' modified here...

**Comment by FredLL-Avaiga** [comment]
> I am very surprised not to see 'dialog.md' modified here...

Un oubli !

**Review by dinhlongviolin1 — APPROVED** [review]
Great job

**Review by FabienLelaquais — APPROVED** [review]
Great!

**Pull request merged**